python/tank/util/user_settings.py

```diff
@@ -45,48 +45,83 @@ def _init_singleton(self):
         logger.debug("Default login: %s" % (self.default_login or "<missing>",))
 
-        self._settings_proxy = self._get_settings_proxy()
-        self._system_proxy = None
-        if self._settings_proxy:
-            logger.debug("Shotgun proxy (from settings): %s" % self._get_filtered_proxy(self._settings_proxy))
-        else:
-            self._system_proxy = self._get_system_proxy()
+        # Retrieve the operating system http proxy string,
+        # or None when it is not defined.
+        self._system_proxy = self._get_system_proxy()
+
+        # Retrieve the settings http proxy string (possibly an empty string),
+        # or None when it is not present in the file.
+        self._settings_proxy = self._get_value("http_proxy")
+
+        if self._settings_proxy is None:
+            # Since the http proxy is not present in the settings file,
+            # fall back on the operating system http proxy.
             if self._system_proxy:
                 logger.debug("Shotgun proxy (from system): %s" % self._get_filtered_proxy(self._system_proxy))
             else:
                 logger.debug("Shotgun proxy: <missing>")
+        else:
+            if self._settings_proxy:
+                logger.debug("Shotgun proxy (from settings): %s" % self._get_filtered_proxy(self._settings_proxy))
+            else:
+                logger.debug("Shotgun proxy (from settings): <empty>")
 
-        proxy = self._get_filtered_proxy(self.app_store_proxy)
-        if self.is_app_store_proxy_set():
-            logger.debug("App Store proxy: %s" % (proxy or "<empty>",))
+        # Retrieve the settings app store proxy string (possibly an empty string),
+        # or None when it is not present in the settings file.
+        self._app_store_proxy = self._get_value("app_store_http_proxy")
+
+        if self._app_store_proxy is None:
+            # Since the app store proxy is not present in the settings file,
+            # fall back on the operating system http proxy.
+            if self._system_proxy:
+                logger.debug("App Store proxy (from system): %s" % self._get_filtered_proxy(self._system_proxy))
+            else:
+                logger.debug("App Store proxy: <missing>")
         else:
-            logger.debug("App Store proxy: <missing>")
+            if self._app_store_proxy:
+                logger.debug("App Store proxy (from settings): %s" % self._get_filtered_proxy(self._app_store_proxy))
+            else:
+                logger.debug("App Store proxy (from settings): <empty>")
+
+    def is_shotgun_proxy_set(self):
+        """
+        :returns: ``True`` if ``http_proxy`` is present in the settings file,
+                  ``False`` otherwise.
+        """
+        return self._settings_proxy is not None
 
     @property
     def shotgun_proxy(self):
         """
-        :returns: The default proxy.
+        :returns: The settings http proxy when it is present and non-empty,
```

Comment:
similar to above, the phrase "The settings http proxy" feels ambigous to me. How about something like this?

```
@property
def shotgun_proxy
"""
The proxy server to use for Shotgun connections.
The UserSettings system will look in several different places for the proxy settings:
- first it will attempt to determine it via the http_proxy setting in the configuration file.
- next it will attempt to determine it from the operating system settings
- if not found, None will be returned.

Please read more about toolkit settings here: xxx
```